lib/healthcheck.js

```diff
@@ -1,17 +1,53 @@
 const fetch = require('r2');
+const { get } = require('lodash');
+const { Router } = require('express');
 
-module.exports = settings => (req, res) => {
-  if (settings.api) {
-    return fetch(`${settings.api}/healthcheck`)
+module.exports = settings => {
+  const router = Router();
+
+  const ping = url => {
+    if (!url) {
+      return false;
+    }
+    return fetch(`${url}/healthcheck`)
       .then(response => {
         if (response.status !== 200) {
+          return { url, status: response.status };
+        }
+        return false;
+      })
+      .catch(() => {
+        return { url, status: null };
+      });
+  };
+
+  router.get('/ready', (req, res, next) => {
+    const services = [
+      settings.api,
+      settings.workflow,
+      settings.notifications,
+      settings.emailer,
```

Comment:
why only these?

Reply:
What do you think is missing?

Reply:
resolver

Reply:
The pdf generator is arguably missing, but a) it's awkward because it's configured with a `/convert` so won't play nicely with the healthcheck, and b) we can run 95% of the service without it, so it's not worth taking the whole thing down if it's not there.

Reply:
ooh, no that's not web facing

Reply:
Actually, having said that I'm going to remove notifications for the same reason.

Reply:
ignore :)